libkbfs/interfaces.go

```diff
@@ -1850,3 +1864,17 @@ type KeyBundleCache interface {
 	// PutTLFWriterKeyBundle stores the given TLFWriterKeyBundleV3.
 	PutTLFWriterKeyBundle(tlf.ID, TLFWriterKeyBundleID, *TLFWriterKeyBundleV3)
 }
+
+// Prefetcher is an interface to a block prefetcher.
+type Prefetcher interface {
+	// PrefetchDirBlock directs the prefetcher to prefetch a directory block.
+	PrefetchDirBlock(blockPtr BlockPointer, kmd KeyMetadata, priority int) error
+	// PrefetchFileBlock directs the prefetcher to prefetch a file block.
+	PrefetchFileBlock(blockPtr BlockPointer, kmd KeyMetadata, priority int) error
+	// HandleBlock allows the prefetcher to determine how to handle a retrieved
+	// block.
+	HandleBlock(b Block, kmd KeyMetadata, priority int)
+	// Shutdown shuts down the prefetcher idempotently. Future calls to
```

Comment:
The returned chan is only for testing, so that tests can be assured that triggered prefetches have completed.